src/components/Heading/Heading.tsx

```diff
@@ -1,56 +1,82 @@
 import clsx from 'clsx';
-import React, { ReactNode } from 'react';
+import React, { forwardRef } from 'react';
 import styles from './Heading.module.css';
 
-export interface Props {
+export const VARIANTS = [
+  'alert',
+  'base',
+  'brand',
+  'inherit',
+  'neutral',
+  'success',
+  'warning',
+  'white',
   /**
-   * Child node(s) that can be nested inside component
-   */
-  children: ReactNode;
+   * @deprecated Info variant is deprecated.
+   */ 'info',
+] as const;
+export type Variant = typeof VARIANTS[number];
+export type HeadingElement = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
```

Comment:
I vaguely remember them specifically bringing up the difference between the EDS sizes vs the BM sizes, but I don't remember if there was a specific reason why they preferred one over the other – maybe because it's easier to insert or remove a size in the middle of the list? I personally like this one a lot better because it's somehow easier for me to remember than single digit numbers.

Reply:
Hmmm I think it may be that more sizes can be made ( >6 ) but heading tags are just `h1` - `h6`.  Which would also explain why the `as` prop is required and not the `size` tag. 
Thinking this, do you think we should go towards that kind of syntax?

Reply:
Some random thoughts:
- Right now the way we handle `size` and `as` works better if they're using the same levels (so you only specify `as` if it's different than `size`). If we changed size to ints, I imagine you would specify both? ie `<Heading as="h4" size={4}>`
- On the other hand, making people specify `as` might make them think harder about what element a heading should actually be.
- I feel like the BM size levels are basically h levels but they just dropped the h because size 4 does correlate to h4, etc. Because they're basically the same, dropping the h seems pretty pointless to me.
- Limiting headings to only have 6 levels doesn't sound like a bad idea.